### ISSUE TITLE
core repos failing in JDK 17 build, related to lambda deserialization

### ISSUE THREAD

**Issue by SethTisue** [scala/community-build#1435]
but it doesn't happen for me locally on MacOS, I'm only seeing it on Jenkins, and only in 2.12.x, not 2.13.x

sample stack trace: https://gist.github.com/SethTisue/7d3f147f5061bae1978a73868f1941a2

@retronym I haven't really dug into this yet, but before I even start digging, does this ring any obvious bells...?

**Comment by retronym** [comment]
https://github.com/openjdk/jdk/commit/fc08af58cb0571ed375a7937aac7a951ba224644#diff-d19e5ac459022586c38dfa9f2b111ac223484586091eaff12a24fb6d417a9506R510

We need to determine ASAP if this is a JDK bug or if we were relying on unspecified behaviour.

**Comment by retronym** [comment]
Found a real, but unrelated bug in `LambdaDeserializer`. The description of that ticket explains the terrain so could be useful for figuring out this one: https://github.com/scala/bug/issues/12418

**Comment by retronym** [comment]
The bug might not repro on typelevel/scalacheck:main because of a bug introduced to the `SerializationTest` [in April](https://github.com/retronym/scalacheck-1/commit/b21aa534135ed3c9d81d3951b51411f254b256a1#r52471760). I'd guess the community build is using an older version (YAY!)

**Comment by SethTisue** [comment]
> I'd guess the community build is using an older version (YAY!)

I don't think so... all the recent runs have used scalacheck commits from June 2021.

**Comment by SethTisue** [comment]
I am now able to reproduce this locally on MacOS. I had been on 17-ea+22-1964 before, but (unless I was merely confused before) upgrading to 17-ea+27-2476 made the bug show up. On Jenkins, we presently have 17-ea+26-2439.
